src/chat/Eliza.hpp

```diff
@@ -0,0 +1,77 @@
+
+// chris busch (c) copyright 1995 all rights reserved
+#pragma once
+
+#include "Database.hpp"
+#include <cstdlib>
+#include <cstring>
+
+constexpr inline auto MaxDatabases = 100;
+constexpr inline auto MaxNamedDatabases = 200;
+constexpr inline auto MaxInputLength = 100;
+
+namespace chat {
+
+class Eliza {
+public:
+    /**
+     * When loading the chat database file, this is used to map the name of a mob
+     * to the index number of a database that contains the keywords that it will
+     * recognise and the list of responses it can use.
+     */
+    struct NamedDatabase {
+        char *name{};
```

Comment:
I think it'd be simple now to make this a std::string and get rid of the nasty. is the database name really mutable?

Reply:
I have put it on my TODO list